crates/fluvio-protocol-derive/src/ast/prop.rs

```diff
@@ -151,76 +143,322 @@ impl UnnamedProp {
         }
     }
 }
-
-pub fn validate_versions(min: i16, max: Option<i16>, field: Option<&str>) -> Option<String> {
-    match (max, field) {
-        // Print name in named fields
-        (Some(max), Some(field)) if min > max => Some(format!(
-            "On {field}, max version({max}) is less than min({min})."
-        )),
-        // No name to print in unnamed fields
-        (Some(max), None) if min > max => {
-            Some(format!("Max version({max}) is less than min({min})."))
-        }
-        (None, Some(field)) if min < 0 => {
-            Some(format!("On {field} min version({min}) must be positive."))
+/// Convert the values to TokenStream which will be ready to use variable value
+///
+/// # Example
+/// ````ignore
+/// // Function as a literal
+/// fn test() -> i16 { 1 }
+/// #[fluvio(min_version = "test()")]
+/// ````
+/// To use the value from the test() function:
+/// ````ignore
+/// let func_value = prop_attrs_type_value(prop_attr_type, None)
+/// ````
+/// To set a specific type you can do this:
+/// ````ignore
+/// let ident_type = Ident::new("u8", Span::call_site());
+/// let func_value = prop_attrs_type_value(prop_attr_type, Some(&ident_type))
+/// ````
+///
+pub fn prop_attrs_type_value(
+    attrs_type: Option<&PropAttrsType>,
+    ident_type: Option<&Ident>,
```

Comment:
This is always `None` except one occurrence which is in tests.

Reply:
Currently yes but we might need this at any point in the future as currently the default is i16, but if there is even a single value that needs to be different than that, we would need this as I can see that there are places in the the proc-macro that are specifically set as `u8` even though they don't affect this specific case for now.

Reply:
In any case, this is hard to read. You are passing `ident_type: Option<&Ident>` which itself is only relevant for only one variant of `PropAttrsType`. 
And if there is no `attrs_type` (None), why do we return `0_i16`

Reply:
I was just following how it worked previously that it defaults to 0 so I set it to return 0 always.

Reply:
About how the type is parsed, let me know if you like this better. I added a new case IntParse, so if it is needed we can use that now instead of passing ident_type.

https://github.com/infinyon/fluvio/blob/f6d081c81a29c03d495aef8fde94a412f9c30f03/crates/fluvio-protocol-derive/src/ast/prop.rs#L164-L173

Reply:
Now clippy check fails. Just remove it for now because it is not used. We will add it later once it's needed.

Reply:
Yep I saw that will delete for now but actually I was thinking we can even make it just Int(data, typ) without the IntParse, so we will always be parsing the type and this should be even better.